docs/details/HA.md

```diff
@@ -0,0 +1,683 @@
+
+    Concatenate these two files to create a PEM file:
+    
+    ```bash
+    cat pmm.crt pmm.key > pmm.pem
+    ```
+    
+6. Create a directory to store HA Proxy configuration.
+    
+    ```bash
+    mkdir -p /path/to/haproxy-config
+    ```
+    
+    Replace `/path/to/haproxy-config` with the path where you want to store your HAProxy configuration.
+    
+7. Create an HAProxy configuration file named `haproxy.cfg.template` in that directory. This configuration tells HAProxy to use the `/v1/leaderHealthCheck` endpoint of each PMM server to identify the leader.
+    
+    ```
+    global
+        log stdout    local0 debug
+        log stdout    local1 info
+        log stdout    local2 info
+        daemon
+    
+    defaults
+        log     global
+        mode    http
+        option  httplog
+        option  dontlognull
+        timeout connect 5000
+        timeout client  50000
+        timeout server  50000
+    
+    frontend http_front
+        bind *:80
+        default_backend http_back
+    
+    frontend https_front
+        bind *:443 ssl crt /etc/haproxy/certs/pmm.pem
+        default_backend https_back
+    
+    backend http_back
+        option httpchk
+        http-check send meth POST uri /v1/leaderHealthCheck ver HTTP/1.1 hdr Host www
+        http-check expect status 200
+        server pmm-server-active-http PMM_ACTIVE_IP:80 check
+        server pmm-server-passive-http PMM_PASSIVE_IP:80 check backup
+        server pmm-server-passive-2-http PMM_PASSIVE2_IP:80 check backup
+    
+    backend https_back
+        option httpchk
+        http-check send meth POST uri /v1/leaderHealthCheck ver HTTP/1.1 hdr Host www
+        http-check expect status 200
+        server pmm-server-active-https PMM_ACTIVE_IP:443 check ssl verify none
+        server pmm-server-passive-https PMM_PASSIVE_IP:443 check ssl verify none
+        server pmm-server-passive-2-https PMM_PASSIVE2_IP:443 check ssl verify none
+    ```
+    
+8. Before starting the HAProxy container, use `sed` to replace the placeholders in `haproxy.cfg.template` with the environment variables, and write the output to `haproxy.cfg`.
+    
+    ```bash
+    sed -e "s/PMM_ACTIVE_IP/$PMM_ACTIVE_IP/g" \
+        -e "s/PMM_PASSIVE_IP/$PMM_PASSIVE_IP/g" \
+        -e "s/PMM_PASSIVE2_IP/$PMM_PASSIVE2_IP/g" \
+        /path/to/haproxy.cfg.template > /path/to/haproxy.cfg    
+    ```
+    
+9. Run the HAProxy container.
+    
+    ```bash
+    docker run -d \
+      --name haproxy \
+      --network pmm-network \
+      -p 80:80 \
+      -p 443:443 \
+      -v /path/to/haproxy-config:/usr/local/etc/haproxy \
+      -v /path/to/certs:/etc/haproxy/certs \
+      haproxy:2.4.2-alpine
+    ```
+    
+    Replace `/path/to/haproxy-config` with the path to the `haproxy.cfg` file you created in step 4, and `/path/to/certs` with the path to the directory containing the SSL certificate and private key. 
```

Comment:
the file haproxy is created on step 8, and the path is created on step 6.    4 is not correct.  Thanks